### PR TITLE
[DOCS] Document create stored script API

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes

* Adds reference documentation for the create stored script API
* Adds an index page for the script API docs
* Updates the search template docs to reference the new API docs

Relates to #71376

### Context

A user was recently confused about the permissions needed to store a [search template](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-template.html). The current search template docs don't contain the privileges of the create stored script API and probably shouldn't.

To fix this, I'm creating reference docs for several APIs referenced in the search template docs, including the create stored script API. To make reviews easier, I'm breaking this work up over several PRs.

### Previews

* Create stored script API ref: https://elasticsearch_71493.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/create-stored-script-api.html
* Scripts APIs index: https://elasticsearch_71493.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/script-apis.html

### CODE REVIEW

File: docs/reference/scripting/apis/create-stored-script-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+`<script-id>`::
+(Required, string)
+Name or identifier for the stored script or search template. Must be unique
+within the cluster.
+
+`<context>`::
+(Optional, string)
+{painless}/painless-contexts.html[Painless context] in which the script runs.
+You can only specify this parameter for Painless scripts.
```
Comment (jdconrad) [inline docs/reference/scripting/apis/create-stored-script-api.asciidoc:52]: Contexts can be applied to any scripting language. It's just that expressions and mustache have a limited subset of contexts they support. As a long term change we should move painless-contexts into script-contexts. (I have spoken with @lockewritesdocs about this recently). This context parameter is used to immediately try to compile a script when it's stored which guarantees that it won't have compile-time errors when it's used for that specific context.

File: docs/reference/scripting/apis/create-stored-script-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+[[create-stored-script-api-query-params]]
+==== {api-query-parms-title}
+
+`context`::
+(Optional, string)
+{painless}/painless-contexts.html[Painless context] in which the script runs.
+You can only specify this parameter for Painless scripts.
```
Comment (jdconrad) [inline docs/reference/scripting/apis/create-stored-script-api.asciidoc:60]: Same as the comment above.

**Review by jdconrad — CHANGES_REQUESTED** [review]
@jrodewig Thanks for adding these. I have added a comment about incorrect information related to contexts.

File: docs/reference/scripting/apis/create-stored-script-api.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/scripting/apis/create-stored-script-api.asciidoc]: Thanks for the explanation @jdconrad. That makes sense. I pushed some commits to fix the wording here and do some light reorg for future planned API docs. We can xref to the context docs here once those are finalized.

**Review by jdconrad — APPROVED** [review]
LGTM. Thanks for the changes.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**